pxtlib/commonutil.ts

```diff
@@ -71,10 +71,15 @@ namespace ts.pxtc.Util {
         return _localizeLang;
     }
 
+    // This function returns normalized language code
+    // For example: zh-CN this returns ["zh-CN", "zh", "zh-cn"]
+    // First two are valid crowdin\makecode locale code,
+    // Last all lowercase one is just for the backup when reading user defined extensions & tutorials.
     export function normalizeLanguageCode(code: string): string[] {
         const langParts = /^(\w{2})-(\w{2}$)/i.exec(code);
         if (langParts && langParts[1] && langParts[2]) {
-            return [`${langParts[1].toLowerCase()}-${langParts[2].toUpperCase()}`, langParts[1].toLowerCase()];
+            return [`${langParts[1].toLowerCase()}-${langParts[2].toUpperCase()}`, langParts[1].toLowerCase(),
+             `${langParts[1].toLowerCase()}-${langParts[2].toLowerCase()}`];
```

Comment:
feels like the order should be ``zn-cn, zn-CN, zn``

Reply:
we shoudl always try regional codes first

Reply:
No the right locale is case sensitive. See https://support.crowdin.com/api/language-codes/

Reply:
zn-CN is the right locale. first part of the locale is lower case and second is capital case.

Reply:
then we should reorder to ``zh-CN, zh-cn, zh``

Reply:
Ok I changed the order for 3rd party extensions & tutorial. Doesn't make sense other places.